controllers/storagecluster/generate.go

```diff
@@ -79,3 +79,7 @@ func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string)
 func generateStorageQuotaName(storageClassName, quotaName string) string {
 	return fmt.Sprintf("ocs-%s-%s", storageClassName, quotaName)
 }
+
+func generateProviderDeploymentName(initData *ocsv1.StorageCluster) string {
+	return fmt.Sprintf("%s-provider-server", initData.Name)
```

Comment:
I think a better name for the deployment would be 'external-endpointorconsumer-endpoint`

Reply:
Let's have more opinions on this. I am listing all 3 names here.

`ocs-storagecluster-external-endpoint`
`ocs-storagecluster-consumer-endpoint`
 `ocs-storagecluster-provider-server`

Reply:
But why d we need `ocs-storagecluster` as a prefix? Even if we have multiple storage clusters we need a single server. So the options should be:
`ocs-external-endpoint`
`ocs-consumer-endpoint`
`ocs-provider-server`

Reply:
correct I missed this.

Reply:
We might want to add `ocs-consumer-server` to the list

Reply:
going with `ocs-provider-server`